oat-sa/tao-calculator-fe
# Feature/TR-5071/Add strategy for correcting expression

### Related to: https://oat-sa.atlassian.net/browse/TR-5071

### Summary

Add an expression corrector, and fix a few issues.

### Details

A feature is added for improving the user's experience: 
- when the expression ends with an operator, trim it
- when the expression contains open parenthesis, auto-close them

The feature is off by default and is activated thanks to a config.

A few issues have been fixed:
- the evaluation of an expression only containing a function name was displaying a javascript code (ex: `cos`)
- when the expression was ending with a function name, it was possible to add an operator despite it is invalid (ex: `tan*`, `3+cos/`)
- wrong nesting of unit tests

### How to test
- run the tests: `npm test`
- launch the sandbox: `npm run dev`
- enter a function, without operand and try evaluating it: it should display `Syntax error` (ex: `sin`)
- enter a function then an operator: th operator should not be added unless it is a sign (ex: `tan` + `*`)
- enter a number, an operator, a function, and finally an operator: the last operator should not be added unless it is a sign (ex: `3+tan` + `*`)
- activate the corrector mode, and evaluate the expressions:
    - `3+`
    - `3*(4+2`
